File: Angular Frontend/src/app/features/services/employee.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { Employee } from '../../core/models/employee';

@Injectable({
  providedIn: 'root'
})
export class EmployeeService {

  private baseUrl: string = `${environment.baseUrl}/employees`;

  constructor(private httpClient: HttpClient) { }

  getEmployeeList() : Observable<Employee[]>{
    return this.httpClient.get<Employee[]>(this.baseUrl);
  }
  
  getEmployeeById(id: number) : Observable<Employee>{
    return this.httpClient.get<Employee>(`${this.baseUrl}/${id}`);
  }

  createEmployee(employee: Employee) : Observable<Employee>{
    return this.httpClient.post<Employee>(this.baseUrl, employee);
  }

  updateEmployee(id: number, employee: Employee) : Observable<Employee>{
    return this.httpClient.put<Employee>(`${this.baseUrl}/${id}`, employee);
  }

  deleteEmployee(id: number): Observable<boolean>{
    return this.httpClient.delete<boolean>(`${this.baseUrl}/${id}`);
  }


}

File: springboot-backend/src/main/java/com/demoapp/springboot/service/EmployeeService.java
package com.demoapp.springboot.service;

import java.util.List;

import com.demoapp.springboot.model.Employee;

public interface EmployeeService {
	
	public List<Employee> getAllEmployees();
	
	public Employee createEmployee(Employee employee);
	
	public Employee getEmployeeById(Long id);
	
	public Employee updateEmployee(Long id, Employee employee);
	
	public boolean deleteEmployee(Long id);

}
